lib/health_check.rb

```diff
@@ -0,0 +1,91 @@
+class HealthCheck
+  include ChefResource
+
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  KEYERROR = 'key error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = nil
```

Comment:
The `overall` method ensures that we will never hit a nil pointer error when trying to reference `@status`, as well as ensuring that it has a value.  erchef and postgres are assumed to be reachable unless exceptions are thrown while trying to communicate with them, at which point we set the appropriate status.  Rest assured that the decision to make `@status` a `nil` value and not the others was not arbitrary.